### ISSUE TITLE
[bitnami/rabbitmq] Authentication no longer working in 8.30.1

### ISSUE THREAD

**Issue by BSchwetzel** [bitnami/charts#9458]
### Name and Version

bitnami/rabbitmq 8.30.1

### What steps will reproduce the bug?

Using the values.yaml provided below works fine when using chart version 8.30.0, but won't work in 8.30.1.

The error message on login (of a technical user):

```
2022-03-17 10:49:29.399456+00:00 [warn] <0.1309.0> HTTP access denied: User 'admin' authentication failed with internal error. Enable debug logs to see the real error.
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> supervisor: {local,ldap_pool_sup}
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> errorContext: child_terminated
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> reason: {{case_clause,{ok,{eldap_search_result,[],[],asn1_NOVALUE}}},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{rabbit_auth_backend_ldap,dn_lookup,2,
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{file,"rabbit_auth_backend_ldap.erl"},{line,826}]},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {rabbit_auth_backend_ldap,call_ldap_fun,3,
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{file,"rabbit_auth_backend_ldap.erl"},{line,588}]},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {rabbit_auth_backend_ldap,with_login,5,
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{file,"rabbit_auth_backend_ldap.erl"},{line,540}]},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {rabbit_auth_backend_ldap,'-with_ldap/3-fun-2-',4,
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{file,"rabbit_auth_backend_ldap.erl"},{line,510}]},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {worker_pool_worker,handle_call,3,
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{file,"worker_pool_worker.erl"},{line,96}]},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {gen_server2,handle_msg,2,
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> [{file,"gen_server2.erl"},{line,1046}]},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,236}]}]}
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> offender: [{pid,<0.476.0>},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {id,1},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {mfargs,{worker_pool_worker,start_link,[ldap_pool]}},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {restart_type,transient},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {significant,false},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {shutdown,4294967295},
2022-03-17 10:49:29.399208+00:00 [erro] <0.474.0> {child_type,worker}]
```

### Are you using any custom parameters or values?

```yaml
---
auth:
  erlangCookie: REDACTED
  password: REDACTED
  
ingress:
  annotations:
    cert-manager.io/cluster-issuer: REDACTED
    cert-manager.io/common-name: REDACTED
  enabled: true
  hostname: REDACTED
  tls: true
  certManager: true


replicaCount: 3

service:
  loadBalancerIP: 'REDACTED'
  type: LoadBalancer
  
advancedConfiguration: |-
  %% -*- mode: erlang -*-
  %% ----------------------------------------------------------------------------
  %% RabbitMQ Configuration File.
  %%
  %% Related doc guide: http://www.rabbitmq.com/configure.html. See
  %% http://rabbitmq.com/documentation.html for documentation ToC.
  %% ----------------------------------------------------------------------------
  [
   %% ----------------------------------------------------------------------------
   %% RabbitMQ LDAP Plugin
   %%
   %% Related doc guide: http://www.rabbitmq.com/ldap.html.
   %%
   %% ----------------------------------------------------------------------------
    {rabbitmq_auth_backend_ldap,
      [
        {tag_queries, [
          {administrator, {in_group_nested, "CN=REDACTED,OU=REDACTED,OU=REDACTED,DC=REDACTED,DC=REDACTED,DC=REDACTED"}},
          {management,    {in_group_nested, "CN=REDACTED,OU=REDACTED,OU=REDACTED,DC=REDACTED,DC=REDACTED,DC=REDACTED"}}
        ]},
        {log, false},
        {dn_lookup_attribute,   "sAMAccountName"},
        {dn_lookup_base,        "OU=REDACTED,OU=REDACTED,DC=REDACTED,DC=REDACTED,DC=REDACTED"},
        {group_lookup_base,     "OU=REDACTED,OU=REDACTED,DC=REDACTED,DC=REDACTED,DC=REDACTED"},
        {dn_lookup_bind, {"CN=REDACTED,OU=REDACTED,OU=REDACTED,OU=REDACTED,DC=REDACTED,DC=REDACTED,DC=REDACTED","REDACTED"}}
      ]
    }
  ].


communityPlugins: >-
  https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/releases/download/3.9.0/rabbitmq_delayed_message_exchange-3.9.0.ez

extraConfiguration: |-
  log.exchange = true

extraEnvVars:
  - name: http_proxy
    value: REDACTED
  - name: HTTP_PROXY
    value: REDACTED
  - name: https_proxy
    value: REDACTED
  - name: HTTPS_PROXY
    value: REDACTED

extraPlugins: >-
  rabbitmq_auth_backend_ldap rabbitmq_delayed_message_exchange
  rabbitmq_event_exchange rabbitmq_tracing

image:
  debug: true

ldap:
  enabled: true
  port: 636
  servers:
    - REDACTED
    - REDACTED
  tls:
    enabled: true
  user_dn_pattern: ${username}

metrics:
  enabled: true
  plugins: rabbitmq_prometheus

persistence:
  size: 2Gi
```

### What is the expected behavior?

Employees can login via Management UI, "technical" users can login via the respective client libraries

### What do you see instead?

Neither employees nor technical users can login anymore

**Comment by carrodher** [comment]
Are you facing the issue when upgrading from `8.3.0` to `8.3.1` or is it a totally new deployment of `8.3.1`?

**Comment by BSchwetzel** [comment]
This was an upgrade, I went from 8.29.0 straight to 8.30.1. I then rolled back to 8.29.0, everything worked again. After that I went to 8.30.0, and it still worked just fine. When I then went from 8.30.0 to 8.30.1 it broke again. I rolled back to 8.30.0 after that and it started working again. I'm still running 8.30.0 now.

So basically, it was an upgrade, but no matter whether I came from 8.29.0 or from 8.30.0, it always broke with 8.30.1 and started working again when rolling back to the previous version.

**Comment by carrodher** [comment]
Did you upgrade the Helm chart using the credentials from the first installation? See https://github.com/bitnami/charts/tree/master/bitnami/rabbitmq#upgrading

**Comment by BSchwetzel** [comment]
Yes, and the probes work fine. The upgrade completes successfully, I "just" can't login anymore. It's also not the first upgrade we've done (although I don't know with which version we started, but this cluster is operating for some months now).

Out of curiosity I started to switch the image tags around (given that 8.30.0 uses -r38, while 8.30.1 uses -r55). Sure enough, using 8.30.0 with the -r55 broke, while using 8.30.1 with -r38 worked. So to me it seems like a problem in the image. The current -r56 also has that problem.

So what now? Should I create an additional issue in https://github.com/bitnami/bitnami-docker-rabbitmq?

**Comment by carrodher** [comment]
There are different topics. When upgrading, the upgrade can work fine in terms of the deployed k8s objects but the problem is that passwords are stored in the PVs/PVCs which are not being updated. If you upgrade **without** setting the `--set auth.password=[PASSWORD] --set auth.erlangCookie=[RABBITMQ_ERLANG_COOKIE]` parameters, the RabbitMQ logic is going to generate a new random password that is not going to match with the one stored in the PVs; making the authentication fail.

Can you check that the password stored in the PV is the same one used during the upgrade process? You can obtain them by following the instructions provided by the installation notes (`helm get notes RELEASE_NAME -n NAMESPACE`). This command will provide you some hints to obtain the passwords, something like:
```
    echo "Password      : $(kubectl get secret --namespace default rabbit-rabbitmq -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)"
    echo "ErLang Cookie : $(kubectl get secret --namespace default rabbit-rabbitmq -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)"
```

Those passwords are the one stored in your PV and the ones expected by the app.

> Out of curiosity I started to switch the image tags around (given that 8.30.0 uses -r38, while 8.30.1 uses -r55). Sure enough, using 8.30.0 with the -r55 broke, while using 8.30.1 with -r38 worked. So to me, it seems like a problem in the image. The current -r56 also has that problem.

Regarding revisions (`-rXX`), Bitnami builds and releases container images on a daily basis in order to upgrade the Debian system packages to the latest version, ensuring there are no fixable CVEs. Each daily release increments the revision, but it is not common to include other changes. Apart from that, the first part of the image tag is the version of the application itself, from `8.30.0` to `8.30.1`, the RabbitMQ version receives a new patch update.

**Comment by BSchwetzel** [comment]
Yes, they are exactly what is expected.

What I am saying in the previous comment boils down to: Regardless of the chart version (8.29.0, 8.30.0, 8.30.1), if I use image revision -r38, it works. I can freely switch between the chart versions, as long as I use -r38. As soon as I use any revision that is at least -r55 (-r55, -r58, -r60) in any of the aforementioned chart versions, it breaks by not letting me log in, neither as the default user "user", nor as user I created, nor via LDAP.

Therefore my conclusion stands as: This problem is caused by the image, not by the chart.

Given that it surfaced through this specific chart upgrade I created this issue.

Further testing revealed: -r49 worked fine, -r50 doesn't. The commit (https://github.com/bitnami/bitnami-docker-rabbitmq/commit/cfdfabbf2fa889caf23e726848c745a00c684449) reveals an update of Erlang.

With that information I could dig up the following: https://github.com/rabbitmq/rabbitmq-server/discussions/4281

So, if you set up an instance of this chart with LDAP authentication, it will break as well.

While I am not a fan of discovering breaking changes like this I understand that LDAP authentication might be rather uncommon and not tested as well. For now, we will just use -r49 until a fixed version of RabbitMQ 3.9.x is released.

As nothing is wrong with this chart I will close this issue.
